dds/DCPS/Serializer.h

```diff
@@ -762,6 +767,14 @@ class OpenDDS_Dcps_Export Serializer {
 
   bool peek(ACE_CDR::ULong& t);
 
+  struct RdState {
+    unsigned char align_rshift;
+    size_t rpos;
+  };
+
+  RdState rdstate() const;
+  void rdstate(const RdState& state);
```

Comment:
It could be but that would provide more information than what `DynamicData` needs. There are other data in `Serializer` like the pointer to its message block, write state whereas `DynamicData` only needs the two member variables for the read state.

Reply:
Maybe, but what about when we want to add writing support to `DynamicData`? That would require more of the state that already provided by `Serializer`. What if someone wants to add a member to `Serializer`, but doesn't know to add it to `RdState` as well?

Reply:
The current design is specific to the original goal of `DynamicData`, that is to support reading only. You have a good point that `Serializer` might add more members to its internal reading state. Should a comment in `RdState` struct saying that it must contain all reading-related variables of `Serializer` be okay in this case?

Reply:
Yes, I'm fine with that. For the sake of finishing this PR we can put off any major changes for now.